### ISSUE TITLE
hocr-pdf could recalculate word positions for resized image

### ISSUE THREAD

**Issue by jronallo** [tmbdev/hocr-tools#39]
If I want to make a PDF from an image that is exactly the same dimensions as were used during OCR, then hocr-pdf can do that. But if I want my PDFs to be smaller in file size then one way is to use images that are resized smaller than were used for OCR. Currently using an image that is a different size from that used during OCR puts the words in the wrong place. As long as the aspect ratio of the image is maintained even if it is a different size it ought to be possible to recalculate where to place words in the PDF so that they show up in the correct location.

Is this feature of interest? Is this an issue anyone else has?

**Comment by stweil** [comment]
Yes, that should be possible. It is also something that I have wanted several times, but not enough to spend time on realizing it. Maybe it should be implemented as a separate tool `hocr-rescale` which takes a hOCR file and rescales all positions / sizes.

Beside getting smaller PDF files there is an additional motivation where such a rescaler would help: for some images OCR gives better results when they are enlarged before running the OCR, for example by doubling width and height. Then the resulting hOCR file has to be rescaled to fit the original image.

**Comment by jronallo** [comment]
Thank you for the idea. That worked nicely. I ended up writing some code in Ruby to resize hOCR before using hocr-pdf. I just thought I would share my code in case this is the kind of thing you might like to add to hocr-tools at some point. I don't know how thorough it is but it worked for my case.

https://github.com/NCSU-Libraries/ocracoke/blob/master/app/processing_helpers/hocr_resizer.rb

**Comment by stweil** [comment]
I think we should stick to using Python for hocr-tools. Your Ruby implementation can help nevertheless as an example.

The most general solution would be a `hocr-transform` which not only supports rescaling but also rotation and shifting. There are use cases which need this kind of transformation.

**Comment by warren-bank** [comment]
fwiw, I also wrote a [similar cli utility](https://github.com/warren-bank/node-hocr-resizer) for node.

**Comment by thepowerthatbe** [comment]
I'd have a similar need for a HOCR Resize for the same reason (performing OCR at higher res, combining with lower-res jpg for a smaller sized PDF with higher-res OCR accuracy. Any chance someone could work on this in python?

**Comment by warren-bank** [comment]
For somebody who knows Python.. it should be super simple. The Ruby script (by @jronallo) uses an html dom parser. My Node script doesn't even bother.. it just uses straight-up regex pattern search/replace with a little basic math sprinkled in.
